--- App/FatPacker.pm
+++ App/FatPacker.pm
@@ -311,7 +311,15 @@
       };
     }
 
-    unshift @INC, bless \%fatpacked, $class;
+    if ($] >= 5.006001) {
+      unshift @INC, bless \%fatpacked, $class;
+    } else {
+      my $obj = bless \%fatpacked, $class;
+      unshift @INC, sub {
+        shift; # ignore CODE itself
+        return $obj->INC(@_);
+      };
+    }
   } # END OF FATPACK CODE
   END_END
 }
